src/reactors/V2DutchOrderReactor.sol

```diff
@@ -0,0 +1,142 @@
+import {Permit2Lib} from "../lib/Permit2Lib.sol";
+import {DutchDecayLib} from "../lib/DutchDecayLib.sol";
+import {
+    V2DutchOrderLib,
+    V2DutchOrder,
+    V2DutchOrderInner,
+    CosignedV2DutchOrder,
+    DutchOutput,
+    DutchInput
+} from "../lib/V2DutchOrderLib.sol";
+import {SignedOrder, ResolvedOrder} from "../base/ReactorStructs.sol";
+
+/// @notice Reactor for v2 dutch orders
+/// @dev V2 orders must be cosigned by the specified cosigner to override timings and starting values
+/// @dev resolution behavior:
+/// - If cosignature is invalid or not from specified cosigner, revert
+/// - If inputOverride is 0, then use inner inputs
+/// - If inputOverride is nonzero, then ensure it is less than specified input and replace startAmount
+/// - For each DutchOutput:
+///   - If override is 0, then use inner output
+///   - If override is nonzero, then ensure it is greater than specified output and replace startAmount
+contract V2DutchOrderReactor is BaseReactor {
+    using Permit2Lib for ResolvedOrder;
+    using V2DutchOrderLib for V2DutchOrder;
+    using DutchDecayLib for DutchOutput[];
+    using DutchDecayLib for DutchInput;
+    using ExclusivityOverrideLib for ResolvedOrder;
+
+    /// @notice thrown when an order's deadline is before its end time
+    error DeadlineBeforeEndTime();
+
+    /// @notice thrown when an order's end time is before its start time
+    error OrderEndTimeBeforeStartTime();
+
+    /// @notice thrown when an order's inputs and outputs both decay
+    error InputAndOutputDecay();
+
+    /// @notice thrown when an order's cosignature does not match the expected cosigner
+    error InvalidCosignature();
+
+    /// @notice thrown when an order's input override is greater than the specified
+    error InvalidInputOverride();
+
+    /// @notice thrown when an order's output override is greater than the specified
+    error InvalidOutputOverride();
+
+    constructor(IPermit2 _permit2, address _protocolFeeOwner) BaseReactor(_permit2, _protocolFeeOwner) {}
+
+    /// @inheritdoc BaseReactor
+    function resolve(SignedOrder calldata signedOrder)
+        internal
+        view
+        virtual
+        override
+        returns (ResolvedOrder memory resolvedOrder)
+    {
+        CosignedV2DutchOrder memory cosignedOrder = abi.decode(signedOrder.order, (CosignedV2DutchOrder));
+        _validateOrder(cosignedOrder);
+        V2DutchOrder memory order = cosignedOrder.order;
+        _updateWithOverrides(order);
+
+        resolvedOrder = ResolvedOrder({
+            info: order.inner.info,
+            input: order.inner.input.decay(order.decayStartTime, order.decayEndTime),
+            outputs: order.inner.outputs.decay(order.decayStartTime, order.decayEndTime),
+            sig: signedOrder.sig,
+            hash: order.hash()
+        });
+        resolvedOrder.handleOverride(order.exclusiveFiller, order.decayStartTime, order.exclusivityOverrideBps);
```

Comment:
maybe im missing something but i feel like the override logic is weird…
- first in validateOrder it validates the overrides exist and arent bigger/smaller than the original amount but not their %s
- then it goes through again and overrides the actual amounts of the order with the override values that were provided, without checking that they were big enough
- then it calls `handleOverride` which increases all of the values of the order with one increased by x%

Reply:
i feel like you definitely dont need both updateOverrides and handleOverrides

Reply:
ohh ok theres 2 types of overrides - the first 2 are the newly introduced `cosigner` overrides which define the result of the secondary auction. Agreed we can remove the validation loop

the third bullet is the `filler override` which we already had in xv1, it basically is if some other public market filler that didn't compete in the auction has a significantly better price they can fill it at some fixed rate above the resolved price. I think we might want to remove that for v2 but keeping in anyways

Reply:
overloaded the terminology :D

Reply:
removing the public override for now